lib/sessions.ts

```diff
@@ -85,6 +109,98 @@ export class Sessions {
     });
   }
 
+  /** Parse a JWT and verify the signature, preferring local verification over remote.
+   *
+   * If maxTokenAge is set, remote verification will be forced if the JWT was issued at (based on
+   * the "iat" claim) more than maxTokenAge seconds ago.
+   */
+  async authenticate_jwt(
+    jwt: string,
+    options?: {
+      maxTokenAge?: number; // seconds
+    }
+  ): Promise<AuthenticateResponse> {
+    try {
+      const session = await this.authenticate_jwt_local(jwt, options);
+      return {
+        session,
+        session_jwt: jwt,
+        session_token: "",
+
+        // Pretend that this was a successful API request.
```

Comment:
Don't pretend- return a new/diff type! Feel free to only return the `session` object from this function call instead.

Reply:
Oh good point! I guess we don't actually need compatibility with `authenticate` on this one. I'll do `{session, session_jwt}` here, since I think it'll be useful to have the new JWT if we do go to the network.